### ISSUE TITLE
Does not work with systemd-resolved

### ISSUE THREAD

**Issue by lunik1** [Maroka-chan/VPN-Confinement#11]
If a system is using resolved with [services.resolved.enable](https://search.nixos.org/options?channel=unstable&from=0&size=50&sort=relevance&type=packages&query=resolved), units using VPN confinement will fail to start with
```
podman-prowlarr.service: Failed to set up mount namespacing: /run/resolvconf: No such file or directory
```

It seems `/run/resolvconf` is not created when using systemd-resolved.

**Comment by Maroka-chan** [comment]
VPN confinement works fine with systemd-resolved enabled. The problem is that you're trying to mount `/run/resolvconf`. It's not that the directory isn't created, it's intentionally made inaccessible to services running in the netns.
systemd-resolved is a DNS resolver, and we don't want it to resolve to a nameserver from the hosts default network, or in other words, we don't want it to "leak" the DNS. Therefore, the path `/run/resolvconf` or `/var/run/resolvconf` is made inaccessible to prevent DNS leaks.

Using fully-fledged containers makes what this module tries to accomplish much easier. So while you *can* use this module for your containers, since you're using podman, I would rather recommend you take a look at [gluetun](https://github.com/qdm12/gluetun).

**Comment by lunik1** [comment]
I get the same error with non-containerised services, and it is fixed by disabling resolved, which causes `/run/resolvconf` to exist. It also seems to be fixed if I create an empty directory at `/run/resolvconf`.

**Comment by Maroka-chan** [comment]
Seems like I jumped to conclusions then. It occurs with just `services.prowlarr` and `services.resolved`?
I'll try to replicate it when I have time. Thanks for reporting this :)

**Comment by lunik1** [comment]
It occurs with prowlarr and a custom prowlarr-in-podman service. I haven't tested anything else.

For now creating a systemd tmpfiles rule which ensures `/run/resolvconf` exists seems to be an effective workaround.

**Comment by Maroka-chan** [comment]
Sorry for the wait :x

> VPN confinement works fine with systemd-resolved enabled.
Nvm, I believe I was thinking of networkd when I said this.

I can see that the problem stems from `services.resolved` changing `networking.resolvconf.package` to `pkgs.systemd`, which causes /run/resolvconf to not exist.

a32f3cc15c6786399ea3c5459e41343b029b4605 should solve this issue.
I settled on using tmpfiles to ensure the path exists as per your suggestion. I've explained the reasoning for this choice in the commit.
